storage/src/lib.rs

```diff
@@ -15,30 +15,46 @@
 // along with the aleo-std library. If not, see <https://www.gnu.org/licenses/>.
 
 use dirs::home_dir;
-use std::path::PathBuf;
+use std::{path::PathBuf, sync::Arc};
+use tempfile::TempDir;
 
 /// The directory name for Aleo-related resources.
 const ALEO_DIRECTORY: &str = ".aleo";
 
 /// An enum to define the operating mode of the Aleo node.
-#[derive(Clone, Debug, PartialEq, Eq)]
+#[derive(Clone, Debug)]
 pub enum StorageMode {
     /// The production mode is used for running a node on the Aleo mainnet.
     Production,
     /// The development mode is used for running a node on a local network.
     Development(u16),
     /// The custom mode is used for running a node on custom configurations.
     Custom(PathBuf),
+    /// Test-only ephemeral storage which self-destructs afterwards.
+    Test(Option<Arc<TempDir>>),
```

Comment:
Nit: why is it wrapped in an Option?

Reply:
while I don't recall this being used, we might want to reuse an instance of storage in some fine-grained tests, which this would allow